### PR TITLE
feat: add detailed timeout error

### PR DESCRIPTION
**Pull request by steve-chavez**
Closes https://github.com/supabase/pg_net/issues/169.

When a timeout occurs, now the error message is like:

```
Timeout of 1000 ms reached. Total time: 1000.527000 ms (DNS time: 0.084000 ms, TCP/SSL handshake time: 0.314000 ms, HTTP Request/Response time: 1000.014000 ms)
```

Before it used to be:

```
Timeout was reached
```

### CODE REVIEW

**Review by soedirgo — APPROVED** [review]

File: src/core.c [hunk 1]
```diff
@@ -164,16 +171,101 @@ void delete_expired_responses(char *ttl, int batch_size){
...
+
+  // convert seconds to milliseconds
+  double dns_time_ms       = _dns_time       * 1000;
+  double handshake_time_ms = _handshake_time * 1000;
+  double http_time_ms      = _http_time      * 1000;
+  double total_time_ms     = total           * 1000;
+
+  // build the error message
+  curl_timeout_msg result = {.msg = {}};
+  sprintf(result.msg,
```
Comment (za-arthur) [inline src/core.c:245]: It is safer to call `snprintf` to make sure that the result string won't be larger than `result.msg`.

**Review by za-arthur — COMMENTED** [review]
Thank you for the PR. All tests pass on my laptop, although I had to increase timeout in `time.sleep` call within `test_http_detailed_timeout()` and `test_http_get_succeed_with_gt_timeout()`.

I have couple of suggestions. One is mentioned below. And the second: would it make more sense to add new column `error_detail` to `net._http_response`? That way it would be easier to make queries to the table without using pattern matching.

File: src/core.c (hunk 1 above)
Comment (steve-chavez) [reply src/core.c]: Good call, done. Also went ahead and moved all this logic to a more proper `errors.hs` module.

**Comment by steve-chavez** [comment]
> would it make more sense to add new column error_detail to net._http_response? That way it would be easier to make queries to the table without using pattern matching.

@za-arthur I thought about that too, but the problem is that users will likely ignore the new detail column and the goal is for them to pay attention to these timings.

Additionally, we don't add more details to other error messages, just this one. All the other errors come raw from libcurl. So not much use for the new column.

**Review by za-arthur — APPROVED** [review]
Thank you for the fix. LGTM.

**Pull request merged**